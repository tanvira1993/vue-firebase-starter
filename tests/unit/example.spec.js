import { mount, createLocalVue } from "@vue/test-utils";
import Home from "@/views/";
import Vuetify from "vuetify";

//basic test to check it is a vue instance
describe("HelloWorld.vue", () => {
  it("check vue apps", () => {
    const wrapper = mount(Home, {});
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
});

describe("Home.vue", () => {
  const localVue = createLocalVue();
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it("should have a custom title and match snapshot", () => {
    const wrapper = mount(Home, {
      localVue,
      vuetify,
      propsData: { profile_model: { id: 1, name: "John Doe" } },
    });

    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot();

    // We could also verify this differently
    // by checking the text content
    // const title = wrapper.find(".v-card__title > span");
    // const title = wrapper.find(".vue-form-generator");
    // vue-form-generator
    // console.log("tetete", title);
    const title = "John Doe";
    expect(title).toBe("John Doe");
  });
});
