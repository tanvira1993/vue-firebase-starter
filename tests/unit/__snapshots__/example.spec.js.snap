// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue should have a custom title and match snapshot 1`] = `
<div class="panel-body" profile_model="[object Object]">
  <div class="vue-form-generator">
    <fieldset>
      <div class="form-group disabled readonly field-input"><label for="id-disabled-text-field"><span>ID (disabled text field)</span>
          <!---->
        </label>
        <div class="field-wrap">
          <div class="wrapper"><input id="id-disabled-text-field" type="text" disabled="disabled" readonly="readonly" class="form-control">
            <!---->
          </div>
          <!---->
        </div>
        <!---->
        <!---->
      </div>
      <div class="form-group featured required field-input"><label for="name"><span>Name</span>
          <!---->
        </label>
        <div class="field-wrap">
          <div class="wrapper"><input id="name" type="text" placeholder="Your name" required="required" class="form-control">
            <!---->
          </div>
          <!---->
        </div>
        <!---->
        <!---->
      </div>
      <div class="form-group field-select"><label for="skills"><span>Skills</span>
          <!---->
        </label>
        <div class="field-wrap"><select id="skills" class="form-control">
            <option value="">&lt;Nothing selected&gt;</option>
            <!---->
            <option value="Javascript">Javascript</option>
            <!---->
            <option value="VueJS">VueJS</option>
            <!---->
            <option value="CSS3">CSS3</option>
            <!---->
            <option value="HTML5">HTML5</option>
          </select>
          <!---->
        </div>
        <!---->
        <!---->
      </div>
      <div class="form-group field-input"><label for="e-mail"><span>E-mail</span>
          <!---->
        </label>
        <div class="field-wrap">
          <div class="wrapper"><input id="e-mail" type="email" placeholder="User's e-mail address" class="form-control">
            <!---->
          </div>
          <!---->
        </div>
        <!---->
        <!---->
      </div>
      <div class="form-group field-checkbox"><label for="status"><span>Status</span>
          <!---->
        </label>
        <div class="field-wrap"><input id="status" type="checkbox">
          <!---->
        </div>
        <!---->
        <!---->
      </div>
    </fieldset>
  </div>
</div>
`;
